crates/cairo-addons/src/vm/vm_consts.rs

```diff
@@ -674,11 +675,48 @@ pub fn create_vm_consts_dict(
     identifiers: &HashMap<String, Identifier>,
     ids_data: &HashMap<String, HintReference>,
     ap_tracking: &ApTracking,
+    constants: &HashMap<String, Felt252>,
+    hint_accessible_scopes: &Vec<String>,
     py: Python<'_>,
 ) -> Result<Py<PyVmConstsDict>, DynamicHintError> {
     let ids_dict = PyVmConstsDict { items: HashMap::new(), vm: vm as *mut VirtualMachine };
     let py_ids_dict = Py::new(py, ids_dict)?;
 
+    // Process constants and make them accessible in Python hints
+    // Constants are in the form {"module.name": value} and are accessible if their module
+    // is in the hint_accessible_scopes
+    for (full_name, value) in constants {
+        let parts: Vec<_> = full_name.split('.').collect();
+        let const_name = parts.last().unwrap_or(&"").to_string();
+        let module_path = parts[..parts.len() - 1].join(".");
+
+        // Check if constant is directly accessible from current scope
+        if hint_accessible_scopes.iter().any(|scope| module_path == *scope) {
+            py_ids_dict
+                .borrow_mut(py)
+                .items
+                .insert(const_name.to_string(), value.to_biguint().into_bound_py_any(py)?.into());
+            continue;
+        }
+
+        // Check if constant is accessible through an alias in any accessible scope
+        for scope in hint_accessible_scopes {
```

Comment:
is this case covered by the tests in this PR? it looks like only the directly accessible case is tested

Reply:
It is tested